nexoan/crud-api/db/queryExecutor/Interfaces/executor.go

```diff
@@ -0,0 +1,454 @@
+package interfaces
+
+import (
+	"context"
+	"lk/datafoundation/crud-api/pkg/schema"
+	pb "lk/datafoundation/crud-api/lk/datafoundation/crud-api"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+// Executor is the main parent interface that defines common behavior for all database executors.
+// It provides a unified way to execute queries across different database types.
+//
+// Example Usage:
+//
+//	executor := GetExecutor()
+//	result, err := executor.Execute(ctx, &QueryParameter{
+//		Operation: "get_metadata",
+//		Filters: map[string]interface{}{
+//			"entityId": "123",
+//		},
+//	})
+type Executor interface {
+	// Execute processes a query based on the provided QueryParameter.
+	// It routes the query to the appropriate database implementation.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing operation details
+	//
+	// Returns:
+	//   - interface{}: Result of the operation, type depends on the operation
+	//   - error: Any error that occurred during execution
+	//
+	// Example:
+	//   result, err := executor.Execute(ctx, &QueryParameter{
+	//       Operation: "get_metadata",
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//       },
+	//   })
+	Execute(ctx context.Context, plan *QueryParameter) (interface{}, error)
+}
+
+// QueryParameter defines the structure for query execution parameters.
+// It provides a standardized way to specify operation details across different database types.
+//
+// Example:
+//
+//	param := &QueryParameter{
+//		Operation: "get_metadata",
+//		Filters: map[string]interface{}{
+//			"entityId": "123",
+//			"kind": "Person",
+//		TimeRange: struct{
+//			StartTime: "2024-01-01T00:00:00Z",
+//			EndTime: "2024-12-31T23:59:59Z",
+//		},
+//	}
+type QueryParameter struct {
+	// Operation specifies the type of operation to perform
+	// Valid values include:
+	// - "get_metadata": Retrieve entity metadata
+	// - "create_entity": Create a new entity
+	// - "get_relationships": Get entity relationships
+	Operation string
+
+	// Filters specify criteria for the operation
+	// Common filter keys:
+	// - "entityId": ID of the entity
+	// - "kind": Entity kind/type
+	// - "name": Entity name
+	Filters map[string]interface{}
```

Comment:
```suggestion
	Parameters map[string]interface{}
```

I think `Filter` name doesn't make sense with a Create or Update operation.

Reply:
ok